src/ImageSharp/Memory/Allocators/MemoryAllocator.cs

```diff
@@ -89,13 +101,42 @@ public virtual void ReleaseRetainedResources()
         where T : struct
         => MemoryGroup<T>.Allocate(this, totalLength, bufferAlignment, options);
 
-    internal static void MemoryGuardMustBeBetweenOrEqualTo(int value, int min, int max, string paramName)
+    internal static void MemoryGuardMustBeBetweenOrEqualTo<T>(int value, int min, int max, string paramName)
+        where T : struct
     {
-        if (value >= min && value <= max)
+        int typeSizeInBytes = Unsafe.SizeOf<T>();
+        long valueInBytes = value * typeSizeInBytes;
+
+        // If a sufficiently large value is passed in, the multiplication will overflow.
+        // We can detect this by checking if the result is less than the original value.
+        if (valueInBytes < value && value > 0)
+        {
+            valueInBytes = long.MaxValue;
+        }
+
+        if (valueInBytes >= min && valueInBytes <= max)
         {
             return;
         }
 
-        throw new InvalidMemoryOperationException($"Parameter \"{paramName}\" must be between or equal to {min} and {max}, was {value}");
+        throw new InvalidMemoryOperationException($"Parameter \"{paramName}\" must be between or equal to {min} and {max}, was {valueInBytes}");
```

Comment:
For `value>max`, it would be nice to have an exception message that is more indicative of a huge allocation attempt.

Reply:
I dunno, they get a stack trace, they know it's from an allocation attempt.